apps/astarte_realm_management/lib/astarte_realm_management/queries.ex

```diff
@@ -558,78 +558,55 @@ defmodule Astarte.RealmManagement.Queries do
     Repo.fetch_all(query, prefix: keyspace, consistency: :quorum)
   end
 
-  def delete_devices_with_data_on_interface(client, interface_name) do
-    devices_statement = "DELETE FROM kv_store WHERE group=:group_name"
+  def delete_devices_with_data_on_interface(realm_name, interface_name) do
+    keyspace = Realm.keyspace_name(realm_name)
+    group_name = "devices-with-data-on-interface-#{interface_name}-v0"
 
-    devices_query =
-      DatabaseQuery.new()
-      |> DatabaseQuery.statement(devices_statement)
-      |> DatabaseQuery.put(:group_name, "devices-with-data-on-interface-#{interface_name}-v0")
-      |> DatabaseQuery.consistency(:each_quorum)
+    query = from KvStore, where: [group: ^group_name]
 
-    with {:ok, _result} <- DatabaseQuery.call(client, devices_query) do
-      :ok
-    else
-      {:error, reason} ->
-        _ = Logger.warning("Database error: #{inspect(reason)}.", tag: "db_error")
-        {:error, :database_error}
-    end
+    _ = Repo.delete_all(query, prefix: keyspace, consistency: :each_quorum)
+
+    :ok
   end
 
   def delete_values(
-        client,
+        realm_name,
         device_id,
         %InterfaceDescriptor{
           interface_id: interface_id,
           storage_type: :multi_interface_individual_properties_dbtable,
           storage: table_name
-        } = _interface_descriptor,
-        _realm_name
+        } = _interface_descriptor
       ) do
-    delete_values_statement = """
-    DELETE
-    FROM #{table_name}
-    WHERE device_id=:device_id AND interface_id=:interface_id
-    """
+    keyspace = Realm.keyspace_name(realm_name)
 
-    delete_query =
-      DatabaseQuery.new()
-      |> DatabaseQuery.statement(delete_values_statement)
-      |> DatabaseQuery.put(:device_id, device_id)
-      |> DatabaseQuery.put(:interface_id, interface_id)
-      |> DatabaseQuery.consistency(:each_quorum)
+    query =
+      from "#{table_name}",
```

Comment:
nit: avoid interpolation of a single string

```suggestion
      from table_name,
```